lib/services/elasticsearch.js

```diff
@@ -645,77 +660,84 @@ class ElasticSearch extends Service {
       updatedAt: dateNow
     };
 
-    // set missing index & type if possible and add metadata
-    let lastAction; // NOSONAR
-    // Declaring "i" inside "for" statements downgrades
-    // performances by a factor of 3 to 4
-    // Fixed in Node.js v8.x and up
-    let i; // NOSONAR
-    for(i = 0; i < esRequest.body.length; i++) {
-      const item = esRequest.body[i];
-      const action = Object.keys(item)[0];
-
-      if (actionNames.indexOf(action) !== -1) {
-        lastAction = action;
-
-        if (!item[action]._type && esRequest.type) {
-          item[action]._type = esRequest.type;
-        }
+    return this.client.indices.getMapping().then(mappings => {
+      // Keep a local cache of existents index/collections on elasticsearch
+      const esCache = Object.keys(mappings).reduce((indexes, index) => {
+        return Object.assign(indexes, { [index]: Object.keys(mappings[index].mappings) });
+      }, {});
+
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove curly braces and "return" from this arrow function body. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://sonarqube.kaliop.net/coding_rules#rule_key=javascript%3AS3524)